### ISSUE TITLE
Event Order

### ISSUE THREAD

**Issue by sammarland** [pouchdb/pouchdb#3645]
For small updates I receive the complete event before I receive the active event. Is this supposed to happen?

**Comment by nolanlawson** [comment]
I don't think it's defined which one comes first, so yeah. Is it causing an issue?

**Comment by sammarland** [comment]
I want to show a spinner when updates are being transferred. If the complete event arrives after the active event then my spinner will continue spinning until the next sync end.

**Comment by nolanlawson** [comment]
Now that I think about it, the complete event isn't even supposed to fire when you're doing live+retry replication. That's odd.

**Comment by sammarland** [comment]
Let me rephrase the question quickly. What's the best way to show a progress indicator whilst we're using live+retry? For a bit of context, this is being used with AngularJS and Ionic.

There may be a bug here that I'm happy to do some more investigation around, but what I'm more interested in is how I'm suppose to solve the above problem.

**Comment by nolanlawson** [comment]
Ah okay. I answered that recently here : http://stackoverflow.com/a/29143448 

Does that help?

**Comment by daleharvey** [comment]
Answer in stack overflow, and for completion, to show a spinner when using live + retry, the `paused` and `active` events are what should be needed, there are bugs with them currently but should be good enough
